src/codegen.rs

```diff
@@ -1233,6 +1326,45 @@ impl<'ctx> CodeGen<'ctx, '_> {
 
                 self.builder.position_at_end(next_block);
             }
+            DMIR::ListCheckSizeDeopt(list, index, deopt) =>{
+                let list_struct = self.emit_read_value_location(list);
+                let index_struct = self.emit_read_value_location(index);
+
+                let index_meta = self.emit_load_meta_value(index_struct);
+
+
+                let index_f32 = self.emit_to_number_or_zero(func, index_meta).data.into_float_value();
```

Comment:
Should be bitcast data to float and then float conversion instead of checking for number tag